third_party/mlir-emitc/lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -526,14 +526,16 @@ bool CppEmitter::hasBlockLabel(Block &block) {
 }
 
 LogicalResult CppEmitter::emitAttribute(Location loc, Attribute attr) {
-  auto printInt = [&](APInt val, bool isSigned) {
+  auto printInt = [&](APInt val, bool isUnsigned) {
     if (val.getBitWidth() == 1) {
       if (val.getBoolValue())
         os << "true";
       else
         os << "false";
     } else {
-      val.print(os, isSigned);
+      SmallString<128> strValue;
```

Comment:
nit: 128 seems pretty big for numbers that are unlikely to be more than a ~20 characters at most :)

Reply:
We've taken this over from printing float values, but you're indeed right. We should consider to choose a smaller string value here.